subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -678,6 +678,25 @@ static int tls_opt_ciphersuite_list_get(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_ciphersuite_used_get(struct net_context *context,
+					void *optval, socklen_t *optlen)
+{
+	const char *ciph;
+
+	if (*optlen != sizeof(int)) {
+		return -EINVAL;
+	}
+
+	ciph = mbedtls_ssl_get_ciphersuite(&context->tls->ssl);
+	if (ciph == NULL) {
```

Comment:
we usually do "if (!ciph)" in zephyr

Reply:
actually we should avoid (!ciph) and keep it as the original. We will have to go through the code and change how it is done so far (MISRA-C: STMT.COND.NOT_BOOLEAN)